src/main/java/fi/okm/jod/yksilo/dto/profiili/SuosikkiDto.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024 The Finnish Ministry of Education and Culture, The Finnish
+ * The Ministry of Economic Affairs and Employment, The Finnish National Agency of
+ * Education (Opetushallitus) and The Finnish Development and Administration centre
+ * for ELY Centres and TE Offices (KEHA).
+ *
+ * Licensed under the EUPL-1.2-or-later.
+ */
+
+package fi.okm.jod.yksilo.dto.profiili;
+
+import fi.okm.jod.yksilo.domain.SuosikkiTyyppi;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+public record SuosikkiDto(
+    @Schema(
+            name = "id",
+            requiredMode = Schema.RequiredMode.NOT_REQUIRED,
+            accessMode = Schema.AccessMode.READ_ONLY)
+        Optional<UUID> id,
```

Comment:
Unnecessary use of optional (a nullable field works here equally well and avoids extra ceremony (Optional.ofNullable(..)) on usage sites.

+1 for using Schema.accessMode, we should apply it consistently to other readonly properties (ids) as well.